rest/route/host_agent.go

```diff
@@ -663,6 +683,15 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 			continue
 		}
 
+		err = task.SetNumNextTaskDispatches(nextTask.Id, nextTask.Execution, nextTask.NumNextTaskDispatches+1)
```

Comment:
Does this get reset back to 0 when the task restarts?

Reply:
Oh I was referring to whether this count gets reset back to 0 if you restart the task. When restarting a task, we basically make a copy of the task and [reset some fields back to a clean state](https://github.com/evergreen-ci/evergreen/blob/21c2cae10e46d135e26b1dcd771add7a4c09658a/model/task/task.go#L2232). So for example, if the number of dispatches is 3 for this task execution, then when the user restarts it, we should make sure that this count resets back to 0 dispatches for the next execution.